src/parser.js

```diff
@@ -17,12 +18,21 @@ function parser(context) {
     const { pos, posMax, src } = state;
 
     if (state.env.bib === undefined) {
-      if (alwaysReloadFiles === true) {
+      if (bibPath !== null && alwaysReloadFiles === true) {
         const { bibData, citeproc } = context.loadFiles(context.options);
         context.bibData = bibData;
         context.citeproc = citeproc;
       }
 
+      if (bibPath === null) {

```

Comment:
Should this not be `if (bibContents !== null)`? Should the bibPath always take precedence over the custom bibContents or should the more specific case win (i.e. bibPath as generic, bibContents as override)?

I think the latter.